src/engagement_db_to_coda/engagement_db_to_coda.py

```diff
@@ -0,0 +1,184 @@
+    # If we find one, move this message to the correct dataset and return.
+    ws_scheme = coda_config.ws_correct_dataset_code_scheme
+    for label in coda_message.get_latest_labels():
+        if label.scheme_id == ws_scheme.scheme_id:
+            ws_code = ws_scheme.get_code_with_code_id(label.code_id)
+            correct_dataset = coda_config.get_dataset_config_by_ws_code_string_value(ws_code.string_value).engagement_db_dataset
+
+            # Clear the labels and correct the dataset (the message will sync with the new dataset on the next sync)
+            # TODO: There is a risk of creating an infinite update loop here if there is a cycle of WS-correction
+            #       in the coda dataset. This needs to be addressed before we can enter production.
+            log.debug(f"WS correcting from {db_message.dataset} to {correct_dataset}")
+            db_message.labels = []
+            db_message.dataset = correct_dataset
+
+            origin_details = {"coda_dataset": coda_dataset_config.coda_dataset_id,
+                              "coda_message": coda_message.to_firebase_map()}
+            engagement_db.set_message(
+                message=db_message,
+                origin=HistoryEntryOrigin(origin_name="Coda -> Database Sync (WS Correction)", details=origin_details),
+                transaction=transaction
+            )
+
+            return
+
+    # We didn't find a WS label, so simply update the engagement database message to have the same labels as the
+    # message in Coda.
+    db_message.labels = coda_message.labels
+    origin_details = {"coda_dataset": coda_dataset_config.coda_dataset_id,
+                      "coda_message": coda_message.to_firebase_map()}
+    engagement_db.set_message(
+        message=db_message,
+        origin=HistoryEntryOrigin(origin_name="Coda -> Database Sync", details=origin_details),
+        transaction=transaction
+    )
+
+
+@firestore.transactional
+def _sync_message_to_coda(transaction, engagement_db, coda, coda_config, message_id):
+    """
+    Syncs a message from an engagement database to Coda.
+
+    This method:
+     - Writes back a coda id if the engagement db message doesn't have one yet.
+     - Syncs the labels from Coda to this message if the message already exists in Coda.
+     - Creates a new message in Coda if this message hasn't been seen in Coda yet.
+
+    :param transaction: Transaction in the engagement database to perform the update in.
+    :type transaction: google.cloud.firestore.Transaction
+    :param engagement_db: Engagement database to sync from.
+    :type engagement_db: engagement_database.EngagementDatabase
+    :param coda: Coda instance to sync the message to.
+    :type coda: coda_v2_python_client.firebase_client_wrapper.CodaV2Client
+    :param message_id: Id of the message to sync.
+    :type message_id: str
+    """
+    db_message = engagement_db.get_message(message_id, transaction=transaction)
```

Comment:
I see db refers engagement database but consider renaming db_message to engagement_db_message /  engmt_db_msg

Reply:
renamed to engagement_db_message, thanks